consensus/istanbul/backend/backend.go

```diff
@@ -508,34 +491,35 @@ func (sb *Backend) Verify(proposal istanbul.Proposal) (time.Duration, error) {
 	state, err := sb.stateAt(block.Header().ParentHash)
 	if err != nil {
 		sb.logger.Error("verify - Error in getting the block's parent's state", "parentHash", block.Header().ParentHash.Hex(), "err", err)
-		return 0, err
+		return 0, nil, err
 	}
 
 	// Make a copy of the state
 	state = state.Copy()
 
 	// Apply this block's transactions to update the state
-	receipts, _, usedGas, err := sb.processBlock(block, state)
+	receipts, allLogs, usedGas, err := sb.processBlock(block, state)
 	if err != nil {
 		sb.logger.Error("verify - Error in processing the block", "err", err)
-		return 0, err
+		return 0, nil, err
 	}
 
 	// Validate the block
 	if err := sb.validateState(block, state, receipts, usedGas); err != nil {
 		sb.logger.Error("verify - Error in validating the block", "err", err)
-		return 0, err
+		return 0, nil, err
 	}
 
 	// verify the validator set diff if this is the last block of the epoch
 	if istanbul.IsLastBlockOfEpoch(block.Header().Number.Uint64(), sb.config.Epoch) {
 		if err := sb.verifyValSetDiff(proposal, block, state); err != nil {
 			sb.logger.Error("verify - Error in verifying the val set diff", "err", err)
-			return 0, err
+			return 0, nil, err
 		}
 	}
 
-	return 0, err
+	result := &istanbul.BlockConsensusAndProcessResult{SealedBlock: block, Receipts: receipts, Logs: allLogs, State: state}
```

Comment:
I don't think you need to use the full `BlockConsensusAndProcessResult` here.  Specifically, the field `SealedBlock` doesn't make sense.

Consider creating a new type called `BlockProcessResult` with the `Receipts`, `Logs`, and `State` fields and the `BlockConsensusAndProcessResult` type uses that new type.